### ISSUE TITLE
Tools/practices/guidelines for keeping the repo lightweight

### ISSUE THREAD

**Issue by da5nsy** [the-turing-way/the-turing-way#3516]
### Summary

The repo is large (~700mb), and this is a barrier to people who want to clone the repo and work on the project locally, especially if they have a poor internet connection.

### What needs to be done?

In this issue I want to draw together a few different topics that are all related with 2 related goals:

1. Discussing whether we could rewrite the git history to reduce the repo size (for example by removing large binary files such as presentations, pdfs, and large images)
2. Propose guidance for how to limit any further increases to the size of the repo.

#### Rewriting git history

@sgibson91 mentioned in a recent call that we've rewritten git history once before, and gave the impression that it was an unpleasant experience that we should not do again. I think this is because of the large number of people involved in the project, and the large number of live PRs that are active at any time. Did I understand correctly @sgibson91?

#### Tools/practices/guidelines for limiting further increases to the size of the repo

- Assess what types of commits add the most to the repo (I think git-filter-repo includes some useful tools for this)
- Resurrect the image compression bot, which is currently disabled
- Write about how (and why) we can limit bloat in the community handbook
- others? (WIP!)

_Adding breadcrumbs to #3272_

**Comment by RichardJActon** [comment]
Is it worth considering using something like git LFS (alternatives include [DVC](https://dvc.org/), and [lakeFS](https://lakefs.io/) to manage binary assets like images in a way that integrates with git but doesn't lead to repo bloat. They add a bit to the barrier to entry but at the same time this is a very common problem people encounter with git repos so learning a tool to deal with it is potentially a useful thing to introduce people to.

**Comment by sgibson91** [comment]
> we've rewritten git history once before, and gave the impression that it was an unpleasant experience that we should not do again. I think this is because of the large number of people involved in the project, and the large number of live PRs that are active at any time.

Yes. If we rewrite history, every open PR will need to be recreated, and everyone with a fork of the repository will need to update it (likely delete it and re-fork). There is a large potential for work-loss here, and a lot of folks who contribute via the GitHub browser UI will not find this intuitive at all. We also don't have enough "power users" with enough free time to provide an "update" service to walk folk through the situation. So I don't think rewriting git history is a favourable option.

> #### Tools/practices/guidelines for limiting further increases to the size of the repo
> * Assess what types of commits add the most to the repo (I think git-filter-repo includes some useful tools for this)
> * Resurrect the image compression bot, which is currently disabled
> * Write about how (and why) we can limit bloat in the community handbook
> * others? (WIP!)

* CI workflows that fail when certain files are added to the repo?

We already have the [file size checker](https://github.com/the-turing-way/the-turing-way/blob/main/.github/workflows/test-file-size.yml) ([script](https://github.com/the-turing-way/the-turing-way/blob/main/tests/test-file-size.sh))that reports large files

The downside of this approach is that the files will already be committed and pushed by the time CI runs, and the commits would have to _actually be dropped_ (with something like `git reset --hard HEAD~1`) to prevent bloating of the `.git` folder, not just `git revert` (which adds a _new_ commit undoing the previous/specified commit, so the first commit with the large file would still exist).

Pre-commit hooks would solve this _if_ we were a community where everyone was using the git CLI (which we are unfortunately not)
